packages/core/src/controllers/crypto.ts

```diff
@@ -130,6 +131,12 @@ export class Crypto implements ICrypto {
       const symKey = this.getSymKey(topic);
       const message = decrypt({ symKey, encoded });
       const payload = safeJsonParse(message);
+      if (typeof payload.topic !== "undefined" && payload.topic !== topic) {
```

Comment:
```suggestion
      if (typeof payload.topic !== "undefined") {
```
I'm not sure whether `payload.topic !== topic` should be in the first `if`. It makes the second `if` redundant as both conditions are the same.